api/lib/controllers/meeting.js

```diff
@@ -314,16 +366,42 @@ module.exports = {
             },
           },
         },
-        { $unwind: "$meetings" },
-        { $replaceRoot: { newRoot: "$meetings" } },
-        { $sort: { date: -1 } },
-        { $skip: parseInt(skip) },
-        { $limit: parseInt(limit) },
+        {
+          $project: {
+            meetings: {
+              $sortArray: {
+                input: "$meetings",
+                sortBy: { date: -1 },
+              },
+            },
+            count: { $size: "$meetings" },
+            owners: "$meetings.owner",
+          },
+        },
+        {
+          $project: {
+            meetings: {
+              $slice: ["$meetings", sliceStart, sliceEnd],
+            },
+            count: 1,
+            owners: 1,
+          },
+        },
       ];
 
-      const meetings = await User.aggregate(pipeline);
+      const [{ meetings, count, owners }] = await User.aggregate(pipeline);
+
+      const reducedOwners = Object.values(
+        owners.reduce((prev, owner) => {
+          prev[owner._id] = owner;
+
+          return prev;
+        }, {})
+      );
 
-      return res.status(200).send(meetings);
+      return res
+        .status(200)
+        .send({ meetings, count, owners: reducedOwners, filtered });
     } catch (err) {
       return res.status(500).send(err);
     }
```

Comment:
You can actually remove the try catch block around this. The [error wrappers](https://github.com/SocexSolutions/agenda-v2/blob/master/api/lib/util/error-wrapper.js) will catch and log everything.